### PR TITLE
Dialyzer warning fixes PR#1

### PR DESCRIPTION
**Pull request by cernivec**
## What
Fixes some of the dialyzer warnings for the  Blockchain app.
Warning fixed is included in each commit message.

This is the 1st PR, not all the warnings are gone yet.

An extra commit included which removes the unused `identity()` function.

### CODE REVIEW

File: apps/blockchain/lib/blockchain/account.ex [hunk 1]
```diff
@@ -514,7 +514,7 @@ defmodule Blockchain.Account do
       iex> Blockchain.Account.get_storage(updated_state, <<01::160>>, 5)
       {:ok, 9}
   """
-  @spec put_storage(EVM.state(), EVM.address(), integer(), integer()) :: t
+  @spec put_storage(EVM.state(), EVM.address(), integer(), integer()) :: EVM.state | { EVM.state, t, t }
```
Comment (ayrat555) [inline apps/blockchain/lib/blockchain/account.ex]: Something is wrong with this typespec. Why should the method return a tuple with duplicate items? ```{ EVM.state, t, t }```
I think it returns just `EVM.state`

File: apps/blockchain/lib/blockchain/account.ex [hunk 2]
```diff
@@ -514,7 +514,7 @@ defmodule Blockchain.Account do
       iex> Blockchain.Account.get_storage(updated_state, <<01::160>>, 5)
       {:ok, 9}
   """
-  @spec put_storage(EVM.state(), EVM.address(), integer(), integer()) :: EVM.state | { EVM.state, t, t }
+  @spec put_storage(EVM.state(), EVM.address(), integer(), integer()) :: EVM.state
```
Comment (ayrat555) [inline apps/blockchain/lib/blockchain/account.ex]: I'm sorry but tuple is actually needed
```
if return_accounts do
      {updated_state, account, updated_account}
    else
      updated_state
end
```

Comment (cernivec) [reply apps/blockchain/lib/blockchain/account.ex]: return_accounts will always be false in put_storage case won't it? So it won't return a tuple.

Comment (ayrat555) [reply apps/blockchain/lib/blockchain/account.ex]: yes, you're right

**Pull request merged**